devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java

```diff
@@ -80,6 +82,17 @@ private static Configuration classpathConfig(Project project, LaunchMode mode) {
         return project.getConfigurations().getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME);
     }
 
+    private static Configuration extensionClasspathConfiguration(Project project, Configuration classpathConfiguration) {
+        Configuration quarkusModelBuilderConfiguration = project.getConfigurations().findByName(QUARKUS_MODEL_CONFIGURATION);
+        if (quarkusModelBuilderConfiguration != null) {
+            project.getConfigurations().remove(quarkusModelBuilderConfiguration);
+        }
+        return project.getConfigurations().create(QUARKUS_MODEL_CONFIGURATION)
+                .extendsFrom(project.getConfigurations()
+                        .getByName(ConditionalDependenciesEnabler.QUARKUS_EXTENSION_CONFIGURATION_NAME))
+                .extendsFrom(classpathConfiguration);
```

Comment:
Is this making all the extensions (including the transitive ones) direct (top level) dependencies in this configuration?

Reply:
Yes it is

Reply:
This looks suspicious to me. Direct dependencies and their order is significant for version convergence. What was the reason to do it this way?

Reply:
I did this because when we collect deployment dependencies, we only look for first level dependencies. Which does not include conditional dependencies.

Reply:
Ah, I see. To not miss the deployment deps of the conditional deps. Ok, I need to reflect on that.

Reply:
Exactly

Reply:
I don't how we could preserve the resolution order